src/main/scala/org/scalastyle/scalariform/ImportsChecker.scala

```diff
@@ -130,3 +140,275 @@ class ImportGroupingChecker extends ScalariformChecker {
+
+    val CompilationUnit(statements, _) = ast
+    statements.immediateChildren.flatMap { n =>
+      val result = n match {
+        case ImportClause(_, BlockImportExpr(prefix, selectors), _) =>
+          val text = exprToText(prefix.contents)
+          checkImport(text, n.firstToken.offset) ++ checkSelectors(selectors)
+
+        case ImportClause(_, Expr(contents), _) =>
+          val text = exprToText(contents)
+          checkImport(text, n.firstToken.offset)
+
+        case _ =>
+          Nil
+      }
+      lastImport = n
+      result
+    }
+  }
+
+  private def exprToText(contents: List[ExprElement]): String = {
+    contents.flatMap {
+      case GeneralTokens(toks) =>
+        toks.map(_.text)
+
+      case n =>
+        throw new IllegalStateException(s"FIXME: unexpected expr child node $n")
+    }.mkString("")
+  }
+
+  /**
+   * Check that the given import belongs to the current group and is ordered correctly within it.
+   */
+  private def checkImport(str: String, offset: Int): Seq[ScalastyleError] = {
+    val errors = new ListBuffer[ScalastyleError]()
+
+    if (!groups(currentGroup)._2.matcher(str).matches()) {
+      // If a statement doesn't match the current group, there are two options:
+      // - It belongs to a previous group, in which case an error is flagged.
+      // - It belongs to a following group, in which case the group index moves forward.
+      for (i <- 0 to currentGroup - 1) {
+        if (groups(i)._2.matcher(str).matches()) {
+          return Seq(newError(offset, "wrongGroup", str, groups(i)._1, groups(currentGroup)._1))
+        }
+      }
+
+      var nextGroup = currentGroup + 1
+      while (nextGroup < groups.size && !groups(nextGroup)._2.matcher(str).matches()) {
+        nextGroup += 1
+      }
+
+      if (nextGroup == groups.size) {
+        throw new IllegalStateException(s"FIXME: import statement does not match any group: $str")
+      }
+
+      errors ++= checkGroupSeparation(currentGroup, nextGroup, offset)
+      currentGroup = nextGroup
+      lastImportInGroup = null
+    } else {
+      // If the statement is in the same group, make sure there is no empty line between it and
+      // the previous import.
+      errors ++= checkNoSeparator(offset)
+    }
+
+    // Ensure import is in alphabetical order.
+    if (lastImportInGroup != null && compareImports(lastImportInGroup, str) > 0) {
+      errors += newError(offset, "wrongOrderInGroup", str, lastImportInGroup)
+    }
+
+    lastImportInGroup = str
+    errors.toSeq
+  }
+
+  /**
+   * Check that the imports inside a multi-import block are ordered.
+   */
+  private def checkSelectors(selectors: ImportSelectors): Seq[ScalastyleError] = {
+    val ImportSelectors(_, first, others, _) = selectors
+
+    val errors = new ListBuffer[ScalastyleError]()
+    val names = Seq(first.contents.head.tokens.head.text) ++
+      others.map( _._2.contents.head.tokens.head.text)
+
+    var last: String = null
+    for (name <- names) {
```

Comment:
Minor, but you could rewrite this to use `names.sliding(2).forall(compareNames...)`.